### ISSUE TITLE
protocols.msn cockroach

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10653]
|<img alt="moea's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| moea reported|
|-|-|
|Trac ID|trac#861|
|Type|defect|
|Created|2005-01-11 04:34:03Z|

Attachments:

* [msn.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d58/d5843c2634578decb7f59b5452225a8eda6cb8a4/4c1ae1c12c31cf439426934a128a0935dae2994f.patch) (445 bytes) - added by moea on 2005-06-30 22:17:58Z - 
* [test_msn.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d58/d5843c2634578decb7f59b5452225a8eda6cb8a4/a7e6c3144ac1fcb6f54b550f8f67cecba462203d.patch) (576 bytes) - added by moea on 2005-06-30 22:18:32Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__861 861
type__defect defect
reporter__moea moea
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1105418043000000 1105418043000000
changetime__1142117921000000 1142117921000000
version__ 
owner__ 
cc__exarkun cc__moea
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="moea's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moea commented|
|-|-|

```
#!html
<pre>
as of twisted 1.3.0rc1, there is a bug in protocols.msn that sometimes causes
gotMessage() (SwitchboardClient) to be passed some of the message headers as if
they are part of the message body.  i'm not familiar with the idiosyncrasies of
the MSN protocol, but it appears that when a conversation is started on a
switchboard server, clients send a distinct set of headers with a blank body,
usually with the client name and whether logging is enabled.  the mime-type
header, and the logging header (the last two) are always passed as part of the
body.  i have tested this against aMSN, and the latest GAIM port, and the method
is dodgy for both.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
A traffic capture which triggers this bug would be nice.  If you can provide
one, I might be able to fix this.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="moea's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moea commented|
|-|-|

```
#!html
<pre>
patches for t.w.protocols.msn and t.w.test.test_msn
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16250]) Apply patch from moea - correctly recognize capability messages.  Fixes [#10653](https://github.com/twisted/twisted/issues/10653)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
